core-bundle/src/Resources/contao/classes/Backend.php

```diff
@@ -1195,7 +1195,7 @@ public static function getDecodedHostname()
 	{
 		$host = Environment::get('host');
 
-		if (strpos($host, 'xn--') !== 'false')
+		if (strpos($host, 'xn--') !== false)
```

Comment:
WTF? How can this have worked?

Reply:
I guess because the condition always matched, `Idna::decode()` was always executed and it probably returns the domain as is if there is nothing to decode.